### PR TITLE
feat: added button component

### PR DESCRIPTION
**Pull request by sks-21**
Added **button** component
Includes 
- Button variants
- Button sizes
- Solid buttons
- outline buttons
- buttons with icons 
- floating buttons

### CODE REVIEW

File: components/button/button.css [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    background-color: var(--color-gray-dark);
+}
+
+.btn-success {
+    background-color: var(--color-green);
+}
+
+.btn-success:hover {
+    background-color: #15803d;
+}
```
Comment (sushil-mod) [inline components/button/button.css:96]: use color variable

File: components/button/button.css [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+.btn-warning {
+    background-color: var(--color-yellow);
+}
+
+.btn-warning:hover {
+    background-color: #eab308;
+}
```
Comment (sushil-mod) [inline components/button/button.css:104]: same color variable

**Review by sushil-mod — APPROVED** [review]
added some comment ,go through  , good to merge

**Review by munjalSaksham71 — APPROVED** [review]
Make the changes what sushil mentioned and then its good to merge.

**Review by ShreyashMahajan — APPROVED** [review]
The naming of classes is very understandable. Good to merge

**Pull request merged**